### DISCUSSION TITLE
Application Disappearing

### DISCUSSION THREAD

**Discussion by jack-bobs** [gravitational/teleport#14136] [Q&A] [answered]
Hi, 

Is there any way of ensuring that applications don't disconnect from the cluster? Either by making service to keep it connected or not?

Just interested in asking as one of our apps that we were testing on Teleport decided to vanish one day & we don't want to reconnect apps all the time!

**Comment by webvictim** [comment]
If the `app_service` which is providing a given application disconnects for whatever reason, the app will disappear after 10gb minutes or so.

The way to stop that happening is to keep the `app_service` running. Using a service manager like systemd is a good idea: https://goteleport.com/docs/setup/admin/daemon/#step-23-create-a-systemd-unit-file

**Comment by jack-bobs** [comment]
Hi @webvictim, appreciate the answer! 

I sometimes run into an issue with this error: 
System has not been booted with systemd as init system (PID 1). Can't operate. Failed to connect to bus: Host is down

When I try and use systemd to set up Teleport as a service on Linux.

**Reply by webvictim** [reply]
Looks like you're not running on a system that uses systemd. What OS are you running? Is this inside a Docker container maybe?

**Reply by jack-bobs** [reply]
Yes, we are running a docker container. Just trying to get everything to work! I've already got the config ready for the application , but would like to now how to use systemd on a Docker container

**Reply by webvictim** [reply]
If you're running a Docker container you shouldn't need systemd - it's for full systems which need to run multiple processes. Docker is generally only used to run one specific process inside a container - our Teleport container uses `dumb-init` as the process supervisor, for example.

**Reply by jack-bobs** [reply]
So, how would I go about creating a service for the container to keep the application always active?

Do you have any suggestions or packages that would be able to run the command provided in the documentation?

**Reply by jack-bobs** [reply]
Or is it worth using dumb-init & creating the service (like systemd) and running it within dumb-init

**Reply by webvictim** [reply]
As long as the container is running, Teleport should stay connected. If you stop the container, it will disconnect.

How are you starting the container at present?

**Reply by jack-bobs** [reply]
We use Portainer to start, stop and restart our containers. 

Since I can't use systemd as it gives that error, I'm trying to at least create a start-up service that will run the command to join the application to the cluster when it boots up. Ideally, I want those containers hosting our web applications to automatically connect to the cluster without requiring myself to manually join those containers back to the Teleport cluster after a restart.

**Reply by webvictim** [reply]
My overall point is that if you're using a Teleport provided image, it has one job: run Teleport. If the container is running, Teleport is running. If Teleport exits, the container will exit.

So if your apps are disappearing while the container is still running, it points to either a networking issue or a bug.

Having sufficient orchestration to restart the container with the correct config if it exits should be all that you need to do to have your apps stay joined to the cluster.

**Comment by webvictim** [comment, marked as answer]
You've mentioned that your application disappears - is that because the Docker container providing the `app_service` is stopped?

**Reply by jack-bobs** [reply]
Yeah, somehow the application disappears even though the container is up & active.

**Reply by webvictim** [reply]
That's odd. Can you share logs from the container and the Teleport auth/proxy service it's connecting to when that happens?

**Reply by jack-bobs** [reply]
Oh, it seems to be fine this morning when I checked!
